SConstruct

```diff
@@ -238,6 +238,28 @@ preSfm = command("sfm", [skymap, transmissionCurvesTarget],
 env.Depends(preSfm, refcat)
 sfm = {(data.visit, data.ccd): data.sfm(env) for data in sum(allData.values(), [])}
 
+# Sky correction
+def processSkyCorr(dataList):
+    """Generate sky corrections"""
+    visitDataLists = defaultdict(list)
+    for ff in allData:
+        for data in allData[ff]:
+            visitDataLists[data.visit].append(data)
+    preSkyCorr = command("skyCorr", skymap,
+                         getExecutable("pipe_drivers", "skyCorrection.py") + " " + PROC + " " + STDARGS +
+                         " --batch-type=smp --cores=1")
+    nameList = ("skyCorr-%d" % (vv,) for vv in visitDataLists)
+    depList = ([sfm[(data.visit, data.ccd)] for data in visitDataLists[vv]] for vv in visitDataLists)
+    cmdList = (getExecutable("pipe_drivers", "skyCorrection.py") + " " + PROC + " " + STDARGS +
+               " --batch-type=smp --cores=1 --id visit=%d" % (vv,) for vv in visitDataLists)
+    validateList = ([validate(SkyCorrValidation, DATADIR, data.dataId) for data in visitDataLists[vv]] for
+                    vv in visitDataLists)
+    return {vv: command(name, [preSkyCorr] + dep, [cmd] + val) for vv, name, dep, cmd, val in
+            zip(visitDataLists, nameList, depList, cmdList, validateList)}
```

Comment:
Took me a while to parse this line; using kwargs in the call to `command` might help a lot.

Reply:
You should have seen it before I refactored it to `zip` over the various generators!